### PR TITLE
rake task to get list and totals of event with corssref events in bot……h nodes.

### PR DESCRIPTION
**Pull request by kjgarza**
## Description
rake tasks to assess the magnitude of events that have crossref-to-crossref relationships, which were wrongly indexed. See https://github.com/datacite/lupo/issues/351 for more details.

This rake task obtains the list of all the events and saves it into a gist file for quick access.

Criteria or from events is:
subj_id in datacite-crossref,datacite-related events MUST always be a DataCite DOI

## List of General Components affected

ITs a new rake task that triggers a method in the events model

## Status
- [x] Ready for Review

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Non Functional Requirement
- [x] Follows the code style of this project.
- [x] Tests Cover Changes 
- [x] All new and existing tests passed.
- [ ] ~~Documentation~~

### CODE REVIEW

File: lib/tasks/event.rake [hunk 1]
```diff
@@ -71,6 +71,15 @@ namespace :crossref do
   end
 end
 
+namespace :subj_id_check do
+  desc 'checks datacite-crossref,datacite-related events have a DataCite DOI in the subject node'
```
Comment (mfenner) [inline lib/tasks/event.rake]: Please use shorter `desc`

File: lib/tasks/event.rake [hunk 2]
```diff
@@ -71,6 +71,15 @@ namespace :crossref do
...
+  task :check => :environment do
+    cursor = ENV['CURSOR'].to_s.split(",") || [Event.minimum(:id),Event.minimum(:id)]
```
Comment (mfenner) [inline lib/tasks/event.rake]: Please align with how we use `cursor` for events in other places, or use a different term.

File: app/models/event.rb [hunk 3]
```diff
@@ -489,6 +489,37 @@ def access_method
...
+        break unless response.results.results.length > 0
+
+        logger.info "[DoubleCheck] DoubleCheck #{response.results.results.length}  events starting with _id #{response.results.to_a.first[:_id]}."
+        cursor = response.results.to_a.last[:sort]
+
+        # dois = response.results.results.map(&:subj_id)
+        events = response.results.results
+        events.lazy.each do | event|
+          subj_prefix = event.subj_id[/(10\.\d{4,5})/,1]
+          File.open("evens_with_double_crossref_dois.txt", "a+") do |f|
```
Comment (mfenner) [inline app/models/event.rb]: Please don't use `File.open` in a container, as file storage is not persistent. Rather write to another place, e.g. via API call.

File: app/models/event.rb [hunk 4]
```diff
@@ -489,6 +489,37 @@ def access_method
...
+            f.write(event.uuid, "\n")
+            total_errors= total_errors+1
```
Comment (mfenner) [inline app/models/event.rb]: linting...

File: lib/tasks/event.rake (hunk 2 above)
Comment (kjgarza) [reply lib/tasks/event.rake]: done

File: app/models/event.rb (hunk 4 above)
Comment (kjgarza) [reply app/models/event.rb]: done

File: app/models/event.rb (hunk 3 above)
Comment (kjgarza) [reply app/models/event.rb]: Now i pushing the data via API to storage outside the container. However, I am still creating a file to make the whole process faster and send it over a single API call.

**Comment by kjgarza** [comment]
@mfenner Crossref needs a list of the events that have problems to deal with them on their side (flag/erase/hide IDK). The file is the best way to get them the list. Saving the file to a gist it's a fast, simple, and straightforward way to share the list with crossref. The rake task only needs to use this time. 

I could write this rake task into a script and run it from my computer. But the rake task has the advantage of being a speedy way to deliver this to crossref and its easy to manage.

This rake task doesn't have to live permanently in our code. We can remove our repurpose later. We have done that in the past in with, for example, `migrate_landing_page`. See https://github.com/datacite/lupo/blob/ca71f26d5653ae913ae45678349ee2e933e296d9/app/models/doi.rb#L1532

**Comment by mfenner** [comment]
I am not completely convinced, but please move ahead, using gists.

**Comment by kjgarza** [comment]
thanks

**Pull request merged**